# Model Inference

In [1]:
import pandas as pd
import pickle

# Fungsi untuk memuat model
def load_model(model_path):
    # Load the model from the file
    with open(model_path, 'rb') as file:
        model = pickle.load(file)
    return model

# Fungsi untuk memastikan tipe data yang benar
def prepare_data(data):
    # Convert data to DataFrame
    df = pd.DataFrame(data)
    
    # Pastikan tipe data sesuai
    df['wheelbase'] = df['wheelbase'].astype(float)
    df['carlength'] = df['carlength'].astype(float)
    df['carwidth'] = df['carwidth'].astype(float)
    df['curbweight'] = df['curbweight'].astype(int)
    df['enginesize'] = df['enginesize'].astype(int)
    df['boreratio'] = df['boreratio'].astype(float)
    df['horsepower'] = df['horsepower'].astype(int)
    df['citympg'] = df['citympg'].astype(int)
    df['highwaympg'] = df['highwaympg'].astype(int)
    df['CarName_freq'] = df['CarName_freq'].astype(object)
    df['aspiration'] = df['aspiration'].astype(object)
    df['carbody'] = df['carbody'].astype(object)
    df['drivewheel'] = df['drivewheel'].astype(object)
    df['enginelocation'] = df['enginelocation'].astype(object)
    df['enginetype'] = df['enginetype'].astype(object)
    df['cylindernumber'] = df['cylindernumber'].astype(object)
    df['fuelsystem'] = df['fuelsystem'].astype(object)
    
    return df

# Fungsi untuk melakukan prediksi harga mobil
def predict_price(model, data):
    """
    Fungsi untuk memprediksi harga mobil.
    Parameters:
    - model: model yang sudah dilatih dan disimpan
    - data: dictionary berisi fitur-fitur mobil
    
    Returns:
    - prediksi harga mobil
    """
    # Prepare the data
    df = prepare_data(data)
    
    # Predict the price
    pred = model.predict(df)
    return pred

# Contoh penggunaan
if __name__ == "__main__":
    # Path ke model yang disimpan
    model_path = 'random_forest_regressor_model.pkl'
    
    # Load the model
    model = load_model(model_path)
    
    # Contoh data baru untuk prediksi
    data = {
        'wheelbase': [88.6],
        'carlength': [168.8],
        'carwidth': [64.1],
        'curbweight': [2548],
        'enginesize': [130],
        'boreratio': [3.47],
        'horsepower': [111],
        'citympg': [21],
        'highwaympg': [27],
        'CarName_freq': [10],
        'aspiration': ['std'],
        'carbody': ['hatchback'],
        'drivewheel': ['fwd'],
        'enginelocation': ['front'],
        'enginetype': ['ohcf'],
        'cylindernumber': ['four'],
        'fuelsystem': ['mpfi']
    }
    
    # Predict price
    prediction = predict_price(model, data)
    print(f"Predicted price: {prediction[0]}")

Predicted price: 14463.826109668109


Harga pada data baru adalah approx 14464 US Dollar.